src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx

```diff
@@ -23,6 +26,17 @@ function OnboardingModalNavigator() {
     const styles = useThemeStyles();
     const {onboardingIsMediumOrLargerScreenWidth} = useResponsiveLayout();
     const outerViewRef = React.useRef<View>(null);
+    const [accountID] = useOnyx(ONYXKEYS.SESSION, {selector: (session) => session?.accountID ?? 0});
+
+    // Publish a sign_up event when we start the onboarding flow. This should track basic sign ups
+    // as well as Google and Apple SSO.
+    useEffect(() => {
+        if (!accountID) {
+            return;
+        }
+
+        GoogleTagManager.publishEvent(CONST.ANALYTICS.EVENT.SIGN_UP, accountID);
+    }, [accountID]);
```

Comment:
Are there cases when this effect could run multiple times for the user? ie they will get to this point and close the app. They will come back to it later and we will again send the sign_up event. Not sure if its a problem though, just thought about it

Reply:
Yeah if they restart/refresh the app during onboarding, this would get called again. Since it's kind of an edge case we're just ignoring that for now. Also it's not too big of a deal as long as it stays consistent between A/B tests.